xmake/modules/package/tools/autoconf.lua

```diff
@@ -439,17 +439,10 @@ function buildenvs(package, opt)
         -- https://github.com/xmake-io/xmake/issues/2195
         local ld = envs.LD
         if ld and package:has_tool("ld", "clang", "clangxx", "gcc", "gxx") then
-            local dir = path.directory(ld)
-            local name = path.filename(ld)
-            name = name:gsub("clang%+%+$", "ld")
-            name = name:gsub("clang%+%+%-%d+", "ld")
-            name = name:gsub("clang$", "ld")
-            name = name:gsub("clang%-%d+", "ld")
-            name = name:gsub("gcc$", "ld")
-            name = name:gsub("gcc-%d+", "ld")
-            name = name:gsub("g%+%+$", "ld")
-            name = name:gsub("g%+%+%-%d+", "ld")
-            envs.LD = dir and path.join(dir, name) or name
+            local ld = find_tool("ld")
```

Comment:
Using find_tool may find the system ld instead of the ld corresponding to the current package toolchain, which may break many packages, especially for some cross-compilation toolchains.

/sdk/bin/linux-arm-clang -> /sdk/bin/linux-arm-ld

now:

/sdk/bin/linux-arm-clang -> /usr/bin/ld

Reply:
right, maybe i can add the path of the compiler in the search directory list of find_tool, would it work ?

Reply:
no, It is difficult to keep the package toolchain consistent with the result of find_tool. I still recommend that all tool-related information be obtained from package:build_getenv -> package:tool, which can correctly obtain the sdk/bin path corresponding to clang/ld.

Therefore, I currently use the path file name replacement method to obtain ld under the corresponding path.

Reply:
i changed it to only fallback on find_tool if , is it more acceptable ? we can't use package:build_getenv because the point here is that only ld work with autotools, when clang++ is set as ld, it doesn't generate .so for shared libraries